server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java

```diff
@@ -230,16 +230,28 @@ public static GroupRepresentation toGroupHierarchy(GroupModel group, boolean ful
         return rep;
     }
 
-    private static boolean groupMatchesSearchOrIsPathElement(GroupModel group, String search, Boolean exact) {
+    public static boolean groupMatchesSearchOrIsPathElement(GroupModel group, String search, boolean exact) {
         if (StringUtil.isBlank(search)) {
             return true;
         }
-        if(exact !=null && exact.equals(true)){
-            if (group.getName().equals(search)){
+        if (group == null) {
+            return false;
+        }
+
+        String groupName = group.getName();
+        if (exact) {
+            // exact matching applies to the group name only
+            if (groupName.equals(search)){
                 return true;
             }
         } else {
-            if (group.getName().contains(search)) {
+            // fuzzy matching applies to the group name and the group path
+            if (groupName.contains(search)) {
+                return true;
+            }
+
+            String groupPath = ModelToRepresentation.buildGroupPath(group);
```

Comment:
It should be fine to allow searching groups by a path like that. In fact, I think it should be easier when you have multiple groups with the same name within different hierarchies.

I see your point ... Kinda an additional capability (yet another undocumented behavior) being introduced.

@thomasdarimont If we can just update the tests to cover this scenario (search by path), I'm good for it. Do you agree @hmlnarik ?

I could also think about allowing searching by the exact path (instead of using `contains`). Not sure if we should worry about it now.

Reply:
/Search by group path
The search with partial paths like `a/b` is tricky, because the search input is passed down to the database query 
as is within a `...from GroupEntity u where u.realm = :realm and u.name like concat('%',:search,'%') ...` query "getGroupIdsByNameContaining" in GroupEntity.java.
Btw. some queries use `lower(search)` for the comparison but others don't - that seems inconsistent and needs some attention.
See: 
https://github.com/keycloak/keycloak/blob/main/model/jpa/src/main/java/org/keycloak/models/jpa/entities/GroupEntity.java#LL33C27-L33C54

Therefore, if we pass a search query like `a/b` we will never get group results back, because the giben `path` does not match an group name.

A workaround could be to translate a search like `a/b` into a regex pattern like (a|b) and use some "regex like" or dbms specific function to match the group name.

Because of this I would vote to leave out the support for searching by paths for now.

/Exact match with group path elements
FYI I extended the PR with support for exact matching in group names AND path elements.

So with exact=true
a search for "Test" for groups
```
1) /Test/Abc
2) /Test2/Def
3) /Test3/Test/Ghi
```

will match 1) and 3) but not 2).

(need to add a test for the later)

Reply:
> Because of this I would currently vote to leave out the support for searching by paths for now.

I agree.

> So with exact=true a search for "Test" for groups
> 
> ```
> 1) /Test/Abc
> 2) /Test2/Def
> 3) /Test3/Test/Ghi
> ```
> 
> will match 1) and 3) but not 2).
> 
> (need to add a test for the later)

I would expect only `/Test`, and `/Test3/Test` being matched, not  the `Abc`, `Def` and `Ghi`. Am I correct or is it expected to return the children too?